### PR TITLE
Sendgrid dynamic email templates

### PR DESCRIPTION
**Pull request by jzxhuang**
## Description

[Link to ticket](https://www.notion.so/uwblueprintexecs/Building-Up-be1b30c321554bffba37a64c6cdc1ec6?p=fb8dee85d3454e24a484a994ab3f5b11)

Email copies: https://docs.google.com/document/d/1wKz2jnbcEHmtiwzEvC3XYRMF0XzNDlBH6PJcBY6z6Ps/edit

## Approach

You can see the dynamic templates if you sign into SendGrid. I followed https://sendgrid.com/docs/ui/sending-email/how-to-send-an-email-with-dynamic-transactional-templates/#send-a-transactional-email

## Testing

If you can, test out the 4 different emails (verify email, reset password, invite to team, leave team). If you don't have time, please just take a look at the code

## Screenshots

User removed from team:
image

Invite user to team:
image

Verify your email address:
image

Reset password:
image

### CODE REVIEW

File: src/middleware/auth.js [hunk 1]
```diff
@@ -68,11 +68,11 @@ const authenticateToken = (req, res, next) => {
         req.userId = refreshData.userId;
       } catch (error2) {
         // eslint-disable-next-line no-console
-        console.log(error2);
+        // console.log(error2);
```
Comment (hongyi-chen) [inline src/middleware/auth.js]: There's two lingering console.log comments here

File: src/resolvers/auth.js [hunk 2]
```diff
@@ -14,11 +14,16 @@ const {
 } = require('../middleware/auth');
 const { DASHBOARD_ROOT_PATH } = require('../constants/client-routes');
 
-const createResetPasswordEmail = resetToken => {
+const RESET_PASSWORD_TEMPLATE_ID = 'd-b444d727384e40f38efad1175b6681e7';
```
Comment (hongyi-chen) [inline src/resolvers/auth.js:17]: Maybe extract these template IDs to `.env`?

**Review by hongyi-chen — APPROVED** [review]
Tested this and looks really good! left a few small comments but basically good to merge

File: src/resolvers/auth.js (hunk 2 above)
Comment (jzxhuang) [reply src/resolvers/auth.js:17]: maybe will do in the future

**Pull request merged**